lib/puppet/transaction/additional_resource_generator.rb

```diff
@@ -102,16 +103,69 @@ def add_resources(generated, resource)
     end
   end
 
-  def add_resource(res, parent_resource, priority)
+  def add_resource(res, parent_resource, priority=nil)
```

Comment:
yeah, this function is now shared by two different code paths - one that adds just to the catalog (for generate) and one that adds to the relationship graph (eval_generate)